product_stock_state/models/product_product.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2017-Today GRAP (http://www.grap.coop).
+# Copyright 2018 ACSONE SA/NV
+# Copyright 2018 Akretion (http://www.akretion.com).
+# Copyright 2020 Camptocamp SA (http://www.camptocamp.com)
+# @author Sylvain LE GAL <https://twitter.com/legalsylvain>
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# @author Laurent Mignon <laurent.mignon@acsone.com>
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_compare
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    _STOCK_STATE_SELECTION = [
+        ("in_stock", "In Stock"),
+        ("in_limited_stock", "In Limited Stock"),
+        ("resupplying", "Resupplying"),
+        ("out_of_stock", "Out Of Stock"),
+    ]
+
+    stock_state = fields.Selection(
+        selection="_selection_stock_state", compute="_compute_stock_state"
+    )
+
+    def _selection_stock_state(self):
+        return self._STOCK_STATE_SELECTION
+
+    def _get_qty_available_for_stock_state(self):
+        """
+        This method can be overridden to provide the available qty.
+        In some cases you could prefer to use the qty_available - outgoing_qty
+        to take into account products reserved
+        """
+        self.ensure_one()
+        return self.qty_available
+
+    def _stock_state_check_in_stock(self, qty, precision):
```

Comment:
I'm wondering where and why those several private methods are used for ?

Reply:
Removed.

Reply:
@kittiu thanks for that one, I think it's the same for `_stock_state_check_in_limited_stock`, `_stock_state_check_resupplying`, and even more `_stock_state_check_out_of_stock`.

Reply:
Thanks!

Reply:
:+1:

Reply:
@kittiu @petrus-v Why did you remove this method ?
The module is broken now.
Indeed this method are called here : https://github.com/OCA/product-attribute/pull/715/commits/cf2c5a1bd6e9b7fc778466fb9941c6d5a3f5e1ee#diff-a0f343e91cae7060fd8ed792256dc90b6b263767beab71becc4e51be5954002bR55

Reply:
ok I missed that code sorry, but, but... there were not test to prevents removing those methods ;-)

Reply:
hum...  CI his broken probably some of them were covered sorry